app/reflexes/invite_manager_reflex.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+class InviteManagerReflex < ApplicationReflex
+  include ManagerInvitations
+
+  def invite
+    email = params[:email]
+    enterprise = Enterprise.find(params[:enterprise_id])
+
+    authorize! :edit, enterprise
+
+    existing_user = Spree::User.find_by(email: email)
+
+    locals = { error: nil, success: nil, email: email, enterprise: enterprise }
+
+    if existing_user
+      locals[:error] = I18n.t('admin.enterprises.invite_manager.user_already_exists')
+
+      return_morph(locals)
+      return
+    end
+
+    begin
+      new_user = create_new_manager(email, enterprise)
+      locals[:success] = true
+      locals[:email] = new_user.email
+    rescue StandardError => e
+      @error = e.message
+      locals[:error] = @error || I18n.t('admin.enterprises.invite_manager.error')
+    end
```

Comment:
I remember discussing this with Maikel a while back, and agreeing that we shouldn't have a "catch-all" (`rescue StandardError`), and should rescue only expected errors. Any unexpected errors should be treated as bugs (logged in bugsnag) so we can improve them.

I can see that the old controller had a problem: `create_new_manager` could raise an exception (from `new_user.save!`), but it wasn't being caught. 

Handling exceptions in this way is generally a good idea, but for controllers we more commonly use the `if/else` pattern. So I'd recommend we use that. What do others think?

Otherwise, it should be renamed `create_new_manager!` and we should only catch expected errors like `ActiveRecord::RecordInvalid`.

Reply:
My suggestion (untested):
```suggestion
    new_user = create_new_manager(email, enterprise)
    if new_user
      locals[:success] = true
    else
      locals[:error] = new_user.errors.full_messages || I18n.t('admin.enterprises.invite_manager.error')
    end
```
* This requires that `create_new_manager` is changed from `save!` to `save`
* From what I could see `new_user.email` should be the same as local var `email`, so I simplified that too.
* I don't see the need for class var `@error`, so I removed it

Reply:
> This requires that create_new_manager is changed from save! to save

Not only, since `EnterpriseMailer.manager_invitation(@enterprise, new_user).deliver_later` seems to throw exception as well. That's why I've added: 
```ruby
return new_user unless new_user.valid? # Return early if user is invalid.
```

after `new_user.save`

Reply:
Commit: b17d3e447c

Reply:
Great work, thanks for picking that up and making it work.